test/extensions/filters/http/grpc_http1_reverse_bridge/reverse_bridge_integration_test.cc

```diff
@@ -202,5 +208,77 @@ TEST_P(ReverseBridgeIntegrationTest, EnabledRouteBadContentType) {
   ASSERT_TRUE(fake_upstream_connection_->close());
   ASSERT_TRUE(fake_upstream_connection_->waitForDisconnect());
 }
+
+// Verifies that we stream the response instead of buffering it, using an upstream-provided header
+// to get the overall message length.
+TEST_P(ReverseBridgeIntegrationTest, EnabledRouteStreamResponse) {
+  upstream_protocol_ = FakeHttpConnection::Type::HTTP1;
+
+  initialize(absl::optional("custom-response-size-header"));
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  Http::TestRequestHeaderMapImpl request_headers({{":scheme", "http"},
+                                                  {":method", "POST"},
+                                                  {":authority", "foo"},
+                                                  {":path", "/testing.ExampleService/Print"},
+                                                  {"content-type", "application/grpc"}});
+
+  auto response = codec_client_->makeRequestWithBody(request_headers, "abcdef");
+
+  // Wait for upstream to finish the request.
+  ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
+  ASSERT_TRUE(fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_));
+  ASSERT_TRUE(upstream_request_->waitForEndStream(*dispatcher_));
+
+  // Ensure that we stripped the length prefix and set the appropriate headers.
+  EXPECT_EQ("f", upstream_request_->body().toString());
+
+  EXPECT_THAT(upstream_request_->headers(),
+              HeaderValueOf(Http::Headers::get().ContentType, "application/x-protobuf"));
+  EXPECT_THAT(upstream_request_->headers(),
+              HeaderValueOf(Http::CustomHeaders::get().Accept, "application/x-protobuf"));
+
+  // Respond to the request.
+  Http::TestResponseHeaderMapImpl response_headers;
+  response_headers.setStatus(200);
+  response_headers.setContentType("application/x-protobuf");
+  // Trust the response size reported from the upstream. The content-length header won't be present
+  // on responses using chunked encoding, for a real-world example.
```

Comment:
That's a really useful example.  Think it's worth calling out in API docs?

Though curious, if the endpoint knows the size (which it must in order to send the header) would it be possible to not have it chunk and just skip all this complexity altogether?  Largely content length is left out if the sender doesn't know the size of the response, which isn't true in this case.

Reply:
That's a good point. I'm going to confirm that the person asking for this feature doesn't need to chunk.